orchestra/static/orchestra/project/js/controllers.js

```diff
@@ -0,0 +1,870 @@
+      // Prevent initial animation of vis load
+      d3.select('.project-management').style('opacity', 0);
+      $scope.vis = new projectVis($routeParams.projectId, '.project-management .svg-wrapper');
+      $scope.vis.update();
+      d3.select('.project-management').transition()
+        .delay(500)
+        .style('opacity', 1);
+    };
+    $scope.activate();
+
+    function projectVis(projectId, parentSelector) {
+      var vis = this;
+
+      vis.projectId = projectId;
+      vis.parentContainer = d3.select(parentSelector);
+
+      vis.options = {
+        'scaleHeight': 40,
+        'barHeight': 30,
+        'lanePadding': {'top': 30, 'bottom': 25},
+        'marginLeft': 100,
+        'marginRight': 10,
+      }
+
+      vis.relativeTime = true;
+      vis.scaleWidth = 1350;
+
+      vis.crosshair = new crosshair();
+      vis.actions = new actions();
+      vis.utils = new utils();
+      vis.tasks = new tasksVis();
+      vis.assignments = new assignmentsVis();
+      vis.iterations = new iterationsVis();
+
+      var axisWrapper = vis.parentContainer
+        .append('div')
+        .attr('class', 'axis-wrapper')
+
+      axisWrapper.append('svg')
+        .attr({
+          'class': 'x axis',
+          'width': vis.utils.getSvgWidth(),
+          'height': vis.options.scaleHeight,
+        })
+        .style('margin-left', vis.options.marginLeft);
+
+      axisWrapper.append('span')
+        .attr('class', 'x label')
+
+      vis.draw = function() {
+        // vis.drawSvg();
+        vis.setupAxis();
+        vis.tasks.draw();
+        vis.crosshair.draw();
+        vis.parentContainer.style('margin-left', vis.utils.freezePaneWidth() + 'px');
+      }
+
+      vis.update = function() {
+        orchestraApi.projectManagementData(vis.projectId)
+          .then(function(response) {
+            vis.data = response.data;
+            vis.prepareData();
+            dataService.data(vis.data);
+            vis.draw();
+          });
+      }
+
+      vis.prepareData = function() {
+        _.each(vis.data.tasks, function(task) {
+          _.each(task.assignments, function(assignment) {
+            assignment.task = task;
+            _.each(assignment.iterations, function(iteration, i) {
+              iteration.assignment = assignment;
+            })
+          })
+        })
+
+        var timeSortedTasks = _.toArray(vis.data.tasks).sort(function(a, b) {
+                                return d3.ascending(a.start_datetime.toDate(), b.start_datetime.toDate());
+                             });
+        vis.timeSortedSlugs = _.map(timeSortedTasks, function(task) {return task.step_slug})
+      }
+
+      vis.setupAxis = function() {
+        var firstTask = vis.data.tasks[_.first(vis.timeSortedSlugs)];
+        minDatetime = firstTask.start_datetime.toDate();
+
+        var taskEndTimes = _.map(vis.timeSortedSlugs,
+                                 function(slug) {return dataService.taskEnd(dataService.taskFromKey(slug))})
```

Comment:
style: for something this long, it no longer feels like it should be a single-liner---consider putting on multiple lines